builder/views.py

```diff
@@ -118,13 +118,22 @@ def get(self, request: Request, key: str, **kwargs) -> HttpResponse:
 
         return JsonResponse(self._get(request, course))
 
+    @login_required_method(redirect_url=None)
+    def get(self, request: Request, remote_id: int, **kwargs) -> HttpResponse:
+        course = get_object_or_404(Course, remote_id=remote_id)
+
+        if not course.has_access(request, Permission.READ):
+            return JsonResponse({"success": False, "error": f"No access to instance {course.remote_id}"})
+
+        return JsonResponse(self._get(request, course))
+
     @login_required_method(redirect_url=None)
     def post(self, request: Request, key: str, **kwargs) -> HttpResponse:
         if not request.POST:
             return JsonResponse({"success": False, "error": "No POST parameters given"})
 
         if Course.objects.filter(key=key).exists():
-            return HttpResponse("Course already exists", status=400)
+            return HttpResponse(f"Course with key '{key}' already exists in Git manager", status=400)
```

Comment:
Otherwise good but it doesn't make sense for Git Manager to respond with "... in Git manager". Where else would it already exist in when you are making a request to Git Manager? The "Git Manager" part should be added by A+ when showing the error, e.g. "Git Manager error: [git manager response]" or something similar. Printing "[git manager response] in Git Manager" is not good because then changing the error text on Git Manager (we might also want to add other error messages in the future) might also require a change in A+, which goes contrary to the point of service oriented software.